lib/daru/category.rb

```diff
@@ -0,0 +1,738 @@
+module Daru
+  module Category
```

Comment:
Some limitations of Rubocop are just too strict for large math/scientific libraries. We just disable them sometimes :) Like this: https://github.com/lokeshh/daru/blob/9b157342ee1a91efa3bf1a9efcf068113a4f5373/lib/daru/vector.rb#L10

That is mainly about simple metrics, like class/module length.
But when it is about code complexity metrics, like AbcSize, it typically means your code is really to complex and should be simplified or split into several methods. It is not **always** the case (so, from time to time we disable this one too), but your first guess, when you see AbcSize, is trying to simplify/split up the things.